internal/examples/makefile

```diff
@@ -0,0 +1,77 @@
+# Makefile for OpAMP examples
+
+GOCMD?= go
+
+# SRC_ROOT is the top of the source tree (project root)
+SRC_ROOT := $(shell git rev-parse --show-toplevel)
+
+# Build targets
+.PHONY: build-examples
+build-examples: build-example-agent build-example-supervisor build-example-server
```

Comment:
This is good. Can we also remove the duplicates from root makefile? Perhaps we can call into this makefile from root file for `run-examples` and `build-examples` targets.